src/Controller/ControllerFactory.php

```diff
@@ -210,6 +199,28 @@ protected function getActionArgs(Closure $action, array $passedParams): array
         return $args;
     }
 
+    /**
+     * Convert argument to parameter type.
+     *
+     * @param mixed $arg Passed parameter (argument)
+     * @param string|null $typedName Parameter type or null for no type
+     * @return mixed
+     * @throws \InvalidArgumentException When argument cannot be cast to type
+     */
+    protected function castArgumentToType(mixed $arg, ?string $typeName): mixed
+    {
+        if (!$typeName) {
+            return $arg;
+        }
+
+        switch ($typeName) {
+            case 'string':
+                return $arg;
+        }
+
+        throw new InvalidArgumentException("Unable to cast action argument to parameter type `$typeName`.");
```

Comment:
This could probably return `null` and the exception handled by invoke(). I don't think null can be passed from the request, can it?

Reply:
> I don't think null can be passed from the request, can it?

Nope.